src/main/java/net/sf/jabref/gui/shared/OpenSharedDatabaseDialog.java

```diff
@@ -194,6 +197,12 @@ private void applyGlobalPrefs() {
         if (sharedDatabaseUser.isPresent()) {
             userField.setText(sharedDatabaseUser.get());
         }
+
+        if (sharedDatabasePassword.isPresent()) {
```

Comment:
The password should only be set when there is a password available **and** the checkbox to remember the password is checked so please add `&& sharedDatabaseRememberPassword`.